client/controller/web-util.js

```diff
@@ -1,9 +1,18 @@
-import ReactDom from 'react-dom';
+import { createRoot, hydrateRoot } from 'react-dom/client';
+
+export function getRootDomElement() {
+	return document.getElementById( 'wpcom' );
+}
+
+let wpcomRootNode;
 
 export function render( context ) {
-	ReactDom.render( context.layout, document.getElementById( 'wpcom' ) );
+	if ( wpcomRootNode == null ) {
+		wpcomRootNode = createRoot( getRootDomElement() );
+	}
+	wpcomRootNode.render( context.layout );
 }
 
 export function hydrate( context ) {
-	ReactDom.hydrate( context.layout, document.getElementById( 'wpcom' ) );
+	wpcomRootNode = hydrateRoot( getRootDomElement(), context.layout );
```

Comment:
Here you could check if `wpcomRootNode` already exists and throw/report error if it does. Because it should never happen.

Reply:
Great point. I'll create a follow-up PR for it since it changes the behavior of `hydrate` :)